eva-accession-import-automation/create_fasta_from_assembly_report.sh

```diff
@@ -14,36 +14,37 @@ output_folder=$3
 
 exit_code=0
 
+# To ensure the FASTA file exists before running `grep -c`, otherwise it will always fail
+touch ${output_folder}/${assembly_accession}.fa
+
 for genbank_contig in `grep -v -e "^#" ${assembly_report} | cut -f5`;
 do
     echo ${genbank_contig}
 
     # Download each GenBank accession in the assembly report from ENA into a separate file
     # Delete the accession prefix from the header line
-    wget -q -O - "https://www.ebi.ac.uk/ena/browser/api/fasta/${genbank_contig}" | sed 's/ENA|.*|//g' > ${genbank_contig}
+    wget -q -O - "https://www.ebi.ac.uk/ena/browser/api/fasta/${genbank_contig}" | sed 's/ENA|.*|//g' > ${output_folder}/${genbank_contig}
 
     # If a file has more than one line, then it is concatenated into the full assembly FASTA file
     # (empty sequences can't be indexed)
-    lines=`head -n 2 ${genbank_contig} | wc -l`
-    if [ $lines -gt 1 ]
+    lines=`head -n 2 ${output_folder}/${genbank_contig} | wc -l`
+    if [ $lines -le 1 ]
     then
-        cat ${genbank_contig} >> ${output_folder}/${assembly_accession}.fa
-    else
         echo FASTA sequence not available for ${genbank_contig}
         exit_code=1
         continue
     fi
 
-    # Check that an accession is present no more than once in the output FASTA file, otherwise it 
-    # means there are unexpected aliases in the assembly report, which need to be checked
-    accession=`head -n 1 ${genbank_contig} | cut -f1 -d' ' | cut -f1 -d'.' | cut -f2 -d'>'`
+    # Write the sequence associated with an accession to a FASTA file only once
+    accession=`head -n 1 ${output_folder}/${genbank_contig} | cut -f1 -d' ' | cut -f1 -d'.' | cut -f2 -d'>'`
     matches=`grep -c "${accession}" ${output_folder}/${assembly_accession}.fa`
```

Comment:
when I heard about writing only once, I expected another file with just the written contigs instead of the whole sequence, but ok. we can change that if it causes too much delay. I mean, it could mean grepping a fasta 1000s of times

Reply:
Tested it with carrot (ca. 4841 sequences, 352 MB) and it took less than 10 seconds to run to completion, without counting the download time, which seems to be the biggest bottleneck.